buildpack/store.go

```diff
@@ -0,0 +1,38 @@
+package buildpack
+
+import (
+	"path/filepath"
+
+	"github.com/BurntSushi/toml"
+
+	"github.com/buildpacks/lifecycle/launch"
+)
+
+type Buildpack interface {
+	Build(bpPlan Plan, config BuildConfig) (BuildResult, error)
+	ConfigFile() *Descriptor
+	Detect(config *DetectConfig) DetectRun
+}
```

Comment:
What do you think about moving this interface to `descriptor.go`?

Reply:
I think it's better to have the interface declared where it's used (`DirBuildpackStore.Lookup()` returns `Buildpack`) vs. where it's implemented. I have heard this cited as the rule of thumb, but IDK where it came from.

Reply:
I can't explain why but I would prefer it to be next to the `Descriptor` definition.
But I also don't have strong opinion on that so we can leave it where it is right now.